File: binary-number-with-alternating-bits/binary-number-with-alternating-bits.java
class Solution {
    public boolean hasAlternatingBits(int n) {
        String s = Integer.toBinaryString(n);
        
        for(int i = 0; i < s.length() - 1; i++){
            if(s.charAt(i) == s.charAt(i+1)){
                return false;
            }
        }
        return true;
    }
}

File: 693-binary-number-with-alternating-bits/binary-number-with-alternating-bits.cpp
class Solution {
public:
    bool hasAlternatingBits(unsigned n) {
        bitset<32> B(n);
        int mB=31-countl_zero(n);
        bool prevB=1;
        for(int i=mB-1; i>=0; i--, prevB=!prevB){
            if (B[i]==prevB) return 0;
        }
        return 1;
    }
};